src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,462 @@
+using Akka.Actor;
+using Akka.Util.Internal;
+using Neo.ConsoleService;
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NJArray = Newtonsoft.Json.Linq.JArray;
+using NJObject = Newtonsoft.Json.Linq.JObject;
+
+namespace Neo.Plugins
+{
+    public class OracleService : Plugin, IPersistencePlugin
+    {
+        private const int RefreshInterval = 1000 * 60 * 3;
+
+        private readonly ConsoleServiceBase consoleBase = GetService<ConsoleServiceBase>();
+        private NEP6Wallet wallet;
+        private readonly ConcurrentDictionary<ulong, OracleTask> pendingQueue = new ConcurrentDictionary<ulong, OracleTask>();
+        private readonly ConcurrentDictionary<ulong, DateTime> finishedCache = new ConcurrentDictionary<ulong, DateTime>();
```

Comment:
If you are using a lock, I think you should use `Dictionary` instead of `ConcurrentDictionary`.

Reply:
We should use `ConcurrentDictionary`, as rpc request and timer both will modify `pendingQueue` and `finishedCache`.

Reply:
But you are using `lock` syntax, so you have to lock the same object every time you access the dictionary. And once you do, there is no need to use `ConcurrentDictionary`.

Reply:
Done

Reply:
You must lock for all access to the dictionary.

Reply:
I have replaced lock by `ConcurrentDictionary`. It seems a little easier to use.